src/lib/group.ts

```diff
@@ -345,7 +442,14 @@ class Group {
    */
   static check(g: Group) {
     try {
-      Snarky.group.assertOnCurve(g.#toTuple());
+      const { x, y } = g;
+
+      let x2 = x.square();
+      let x3 = x2.mul(x);
+      let ax = x.mul(Pallas.a); // this will obviously be 0, but just for the sake of correctness
+
+      // we also check the zero element (0, 0) here
+      g.#isZero().or(x3.add(ax).add(Pallas.b).equals(y.square())).assertTrue();
```

Comment:
I wonder if in this one place, we should actually check both `x` and `y` to be zero instead of just `x`. or at least carefully think through the implications of leaving `y` unconstrained

Reply:
Good point, I think we definitely should check that otherwise we will be able to create an invalid element with `(0, y)` and still prove, for example, `(0, y) === zero`